src/cpp/core/file_lock/LinkBasedFileLock.cpp

```diff
@@ -277,19 +277,49 @@ Error writeLockFile(const FilePath& lockFilePath)
             proxyPath.absolutePathNative().c_str(),
             lockFilePath.absolutePathNative().c_str());
    
-   // log errors (remove this if it is too noisy on NFS)
+   // detect link failure
    if (status == -1)
    {
-      int errorNumber = errno;
-      
       // verbose logging
+      int errorNumber = errno;
       LOG("ERROR: ::link() failed (errno " << errorNumber << ")" << std::endl <<
           "Attempted to link:" << std::endl << " - " <<
           "'" << proxyPath.absolutePathNative() << "'" <<
           " => " <<
           "'" << lockFilePath.absolutePathNative() << "'");
       
-      LOG_ERROR(systemError(errorNumber, ERROR_LOCATION));
+      // if this failed, we should still make a best-effort attempt to acquire
+      // a lock by creating a file using O_CREAT | O_EXCL. note that we prefer
+      // ::link() since older NFSes provide more guarantees as to its atomicity,
+      // but not all NFS support ::link()
```

Comment:
The error logs I've seen have reported `EOPNOTSUPP` (95) and `EIO` (5), but it seems like the error code returned here is not consistent.